instrumentation/servlet/servlet-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v5_0/response/HttpServletResponseAdviceHelper.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.servlet.v5_0.response;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import io.opentelemetry.javaagent.instrumentation.api.CallDepth;
+import io.opentelemetry.javaagent.instrumentation.api.CallDepthThreadLocalMap;
+
+public class HttpServletResponseAdviceHelper {
+  public static void stopSpan(
+      BaseTracer tracer,
+      Class<?> responseClass,
+      Throwable throwable,
+      Context context,
+      Scope scope,
+      CallDepth callDepth) {
+    if (callDepth.decrementAndGet() == 0 && context != null) {
+      CallDepthThreadLocalMap.reset(responseClass);
```

Comment:
```suggestion
      callDepth.reset();
```
This should work as well, right?

Reply:
Done, but had to first make that method public.